vscode-dotnet-runtime-library/src/Acquisition/DotnetPathFinder.ts

```diff
@@ -0,0 +1,211 @@
+                return dotnetOnRootEmulationPath;
+            }
+            else
+            {
+                this.workerContext.eventStream.post(new DotnetFindPathRootUnderEmulationButNoneSet(`Under emulation but DOTNET_ROOT_X64 is not set.`));
+            }
+        }
+
+        let dotnetOnRootPath = process.env.DOTNET_ROOT;
+        if(EnvironmentVariableIsDefined(dotnetOnRootPath))
+        {
+            // DOTNET_ROOT should be set to the directory containing the dotnet executable, not the executable itself.
+            dotnetOnRootPath = path.join(dotnetOnRootPath!, getDotnetExecutable());
+            this.workerContext.eventStream.post(new DotnetFindPathRootPATHFound(`Found .NET on the root: ${dotnetOnRootPath}`));
+            return dotnetOnRootPath;
+        }
+        return undefined;
+    }
+
+    /**
+     *
+     * @returns A set of the path environment variable(s) for which or where dotnet, which may need to be converted to the actual path if it points to a polymorphic executable.
+     * For example, `snap` installs dotnet to snap/bin/dotnet, which you can call --list-runtimes on.
+     * The 'realpath' of that is 'usr/bin/snap', which you cannot invoke --list-runtimes on, because it is snap.
+     * In this case, we need to use this polymorphic path to find the actual path later.
+     *
+     * In an install such as homebrew, the PATH is not indicative of all of the PATHs. So dotnet may be missing in the PATH even though it is found in an alternative shell.
+     * The PATH can be discovered using path_helper on mac.
+     */
+    public async findRawPathEnvironmentSetting(tryUseTrueShell = true) : Promise<string[] | undefined>
+    {
+        const oldLookup = process.env.DOTNET_MULTILEVEL_LOOKUP;
+        process.env.DOTNET_MULTILEVEL_LOOKUP = '0'; // make it so --list-runtimes only finds the runtimes on that path: https://learn.microsoft.com/en-us/dotnet/core/compatibility/deployment/7.0/multilevel-lookup#reason-for-change
+
+        const searchEnvironment = process.env; // this is the default, but sometimes it does not get picked up
+        const options = tryUseTrueShell && os.platform() !== 'win32' ? { env : searchEnvironment, shell: process.env.SHELL === '/bin/bash' ? '/bin/bash' : '/bin/sh'} : {env : searchEnvironment};
+
+        this.workerContext.eventStream.post(new DotnetFindPathLookupPATH(`Looking up .NET on the path. Process.env.path: ${process.env.PATH}.
+Executor Path: ${(await this.executor?.execute(
+    os.platform() === 'win32' ? CommandExecutor.makeCommand('echo', ['%PATH']) : CommandExecutor.makeCommand('env', []),
+    undefined,
+    false))?.stdout}
+
+Bin Bash Path: ${os.platform() !== 'win32' ? (await this.executor?.execute(CommandExecutor.makeCommand('env', ['bash']), {shell : '/bin/bash'}, false))?.stdout : 'N/A'}
+`
+        ));
+
+        let windowsWhereCommand = '';
+        let unixWhichCommand = '';
+        if(os.platform() === 'win32')

```

Comment:
only one of these two conditions will be true. Could this be simplified as

```
let command = '';

if (os.platform() === 'win32')
{
    command = ...
}
else
{
    command = ...
}

const finderCommand = command; 
```

Reply:
Thank you, this is a good point.